pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -210,6 +221,50 @@ func (r *MachineSyncReconciler) Reconcile(ctx context.Context, req reconcile.Req
 
 // reconcileCAPIMachinetoMAPIMachine reconciles a CAPI Machine to a MAPI Machine.
 func (r *MachineSyncReconciler) reconcileCAPIMachinetoMAPIMachine(ctx context.Context, capiMachine *capiv1beta1.Machine, mapiMachine *machinev1beta1.Machine) (ctrl.Result, error) {
```

Comment:
This function is invoked in two places.
When it is invoked at like [199](https://github.com/openshift/cluster-capi-operator/pull/265/files#diff-a00680d231c0783b5333cc3e2c6f15e625b31572f57b7fb4b598dcb18d2418f3R199)

The `mapiMachine` argument is always an empty machine `mapiMachine := &machinev1beta1.Machine{}` as `mapiMachineNotFound` is `true`.

As such you can't use mapiMachine.Bla() in subsequent calls (line 255 to 265) as they'll all yield empty strings.

Reply:
I'd also probably set the MAPI machine to nil [here](https://github.com/openshift/cluster-capi-operator/pull/265/files#diff-a00680d231c0783b5333cc3e2c6f15e625b31572f57b7fb4b598dcb18d2418f3R168-R169), like we do for the CAPI one [here](https://github.com/openshift/cluster-capi-operator/pull/265/files#diff-a00680d231c0783b5333cc3e2c6f15e625b31572f57b7fb4b598dcb18d2418f3R185-R186).

As later on ([here](https://github.com/openshift/cluster-capi-operator/pull/265/files#diff-a00680d231c0783b5333cc3e2c6f15e625b31572f57b7fb4b598dcb18d2418f3R486)) we check on it to decide whether to update it or to create it.

Reply:
> The mapiMachine argument is always an empty machine mapiMachine := &machinev1beta1.Machine{} as mapiMachineNotFound is true.

>As such you can't use mapiMachine.Bla() in subsequent calls (line 255 to 265) as they'll all yield empty strings.

Good catch, this makes sense! This is for the scenario where the MAPI Machine does not exist, and CAPI is authoritative and we get a failure either converting or fetching CAPI resources.

In this instance, we'll error trying to set the synchronized condition  and add that to the aggregate that'll get logged. I think it likely makes sense to not bother trying to set the condition and just emitting an event?